ironfish/src/wallet/wallet.ts

```diff
@@ -1387,6 +1396,10 @@ export class Wallet {
     this.onAccountRemoved.emit(account)
   }
 
+  async removeAccountSynchronous(account: Account): Promise<void> {
+    await this.walletDb.deleteAccount(account)
```

Comment:
I'm pretty sure this also needs to make some of the same calls in `wallet.removeAccount`. I tried running `wallet:delete --wait default`, and when I ran `wallet:accounts`, the same `default` account still showed up.

Reply:
Thanks for trying it out. I don't understand the event loop too well yet, pushed a version that does the same stuff in `wallet.removeAccount` too. I can't tell if that's safe, though? @dguenther

Reply:
The event loop is just a factor here in order to periodically call `cleanupDeletedAccounts` -- it lets us limit the amount of cleanup done at one time so that it doesn't use too much of the main thread.

You might be asking about updating the database though -- the database is single-process, and database writes done inside of a transaction don't get committed until the end of the transaction, so there's no place where concurrent reads can fetch half-deleted state from the DB.